xml/deployment_firstboot.xml

```diff
@@ -0,0 +1,870 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE chapter PUBLIC "-//Novell//DTD NovDoc XML V1.0//EN" "novdocx.dtd"
+[
+  <!ENTITY % NOVDOC.DEACTIVATE.IDREF "INCLUDE">
+  <!ENTITY % entities SYSTEM "entity-decl.ent">
+  %entities;
+]>
+<chapter id="cha.deployment_firstboot">
```

Comment:
This looks like a DocBook 4 document, but we need DocBook 5. The latter contains the DB namespace, `xml:id` instead of `id` and more.

You need to convert this document into DocBook 5.

1. Preserve your entities first:

        $ sed -i -f /usr/share/daps/lib/entities.preserve.sed xml/deployment_firstboot.xml

2. Convert the DocBook 4 into DocBook 5:

        $ xsltproc --output deployment_firstboot.xml \
           /usr/share/daps/daps-xslt/migrate/suse-upgrade.xsl xml/deployment_firstboot.xml
        $ mv deployment_firstboot.xml xml/

3. Recover your entities:

        $ sed -i -f /usr/share/daps/lib/entities.recover.sed xml/deployment_firstboot.xml

4. Add the above minimal DOCTYPE in the header.

5. Try again. If it works, dance and be happy! :wink:

Reply:
Generally, Toms is right, the script should work. However, this is a relatively simple document and using the script won't be quicker than doing it manually. So in this case, I'd use this alternative:

1. I'd steal the header of another document (say, https://github.com/SUSE/doc-sle/blob/master/xml/apparmor_repositories.xml#L1,L7 -- just as Toms showed in his first comment but including the \<chapter\> tag as well because you need the xmlns:... definitions in your new document.
2. Edit the xml:id in the header -- should be the same as the id of the old document
3. Replace ` id=` -> ` xml:id=` across the document using search/replace.


There is a final step to get it to work with our CI that Toms has not mentioned and that is replacing the `.` characters in _all_ xml:id/linkend/arearefs attributes with `-`. Either do so manually or with this script https://github.com/sknorr/daps-migrate-tools/blob/master/id-dash-change-script
To use the script:

1. Clone the daps-migrate-tools repo
2. Go to your doc-sle repo
3. Run `../daps-migrate-tools/id-dash-change-script xml/deployment_firstboot.xml` without options (fix the path, needs to point at the right repo)

Reply:
Hmn... :thinking: 

> using the script won't be quicker than doing it manually

I'd friendly disagree. :-) I would be cautious to do it manually. Even for "simple" documents.

It would take Carla 1min to copy-and-paste the above commands. As a result, she gets a document which is _DocBook5-ready_. No need to worry about `id` -> `xml:id` changes, missed `ulink` to `link`, DocBook, XInclude, or XLink namespaces etc. That's all done by the stylesheet. So why should anyone do that manually when we already have a working procedure?

I agree with Stefan, two parts are missing: the DOCTYPE declaration and the ID values. Both aren't created automatically (yet). If it would help, I could add these features to the stylesheet too. As a parameter that can be switched on and off. I assume, this is something which would be helpful for future conversions.

So from my perspective, the only thing that Carla has to do after the above procedure is to call Stefan's script.

Maybe this is something we should add to the Hitchhiker's Guide, if not already the case. I haven't checked it yet, still recovering from yesterday.